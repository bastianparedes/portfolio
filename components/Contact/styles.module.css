.linkContainer {
  width: fit-content;
  height: 100%;
  position: absolute;
  top: 0;
  color: rgb(255, 255, 255);
  display: flex;
}

.linksList {
  position: sticky;
  list-style: none;
  bottom: 0;
  margin: auto 0 0;
  padding-top: 4rem;
}

.verticalLine {
  width: 1px;
  height: 10rem;
  background-color: rgb(255, 255, 255);
  margin: 0 auto;
}

@media (max-device-width: 700px) {
  .linkContainer {
    width: fit-content;
    height: fit-content;
    position: relative;
    display: block;
    margin: 0 auto;
    padding: 1rem;
  }

  .linksList {
    padding: 0;
    margin: 0;
    position: relative;
    display: grid;
    grid-template-columns: repeat(4, 1fr);
    gap: 1rem;
  }

  .verticalLine {
    display: none;
  }
}
