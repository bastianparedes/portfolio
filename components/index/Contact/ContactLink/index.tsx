import React from 'react';

import styles from './styles.module.scss';

interface LinkProps {
  link: string;
  children: React.ReactNode;
}

const ContactLink = ({ link, children }: LinkProps): JSX.Element => {
  return (
    <li className={styles.link}>
      <a
        className={styles.link__a}
        href={link}
        target="_blank"
        rel="noreferrer noopener"
      >
        {children}
      </a>
    </li>
  );
};

export default ContactLink;
