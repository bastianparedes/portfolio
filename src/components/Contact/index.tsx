import React from 'react'
import styles from './styles.module.css'
import { AiOutlineLinkedin } from 'react-icons/ai'
import { FiInstagram } from 'react-icons/fi'
import { VscGithubAlt, VscMail } from 'react-icons/vsc'

interface LinkProps {
  link: string
  children?: React.ReactNode
}

const Link = ({ link, children }: LinkProps): JSX.Element => {
  return (
    <li className={styles.link}>
      <a
        className={styles.link__a}
        href={link}
        target="_blank"
        rel="noreferrer noopener"
      >
        {children}
      </a>
    </li>
  )
}

const Contact = (): JSX.Element => {
  return (
    <div className={styles.linkContainer}>
      <ul className={styles.linksList}>
        <Link link="https://github.com/BastianParedes">
          <VscGithubAlt />
        </Link>
        <Link link="mailto:bastian.p.trabajo@outlook.com">
          <VscMail />
        </Link>
        <Link link="https://www.linkedin.com/in/bastianparedes/">
          <AiOutlineLinkedin />
        </Link>
        <Link link="https://www.instagram.com/bastian.p.95/">
          <FiInstagram />
        </Link>
        <div className={styles.verticalLine}></div>
      </ul>
    </div>
  )
}

export default Contact
