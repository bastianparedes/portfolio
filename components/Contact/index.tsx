import React from 'react'
import styles from './styles.module.scss'
import { AiOutlineLinkedin } from 'react-icons/ai'
import { FiInstagram } from 'react-icons/fi'
import { VscGithubAlt, VscMail } from 'react-icons/vsc'
import constants from '../../config/constants.json'
import ContactLink from './ContactLink'

const Contact = (): JSX.Element => {
  return (
    <div className={styles.linkContainer}>
      <ul className={styles.linksList}>
        <ContactLink link={constants.CONTACT.LINKS.GITHUB}>
          <VscGithubAlt />
        </ContactLink>
        <ContactLink link={constants.CONTACT.LINKS.EMAIL}>
          <VscMail />
        </ContactLink>
        <ContactLink link={constants.CONTACT.LINKS.LINKEDIN}>
          <AiOutlineLinkedin />
        </ContactLink>
        <ContactLink link={constants.CONTACT.LINKS.INSTAGRAM}>
          <FiInstagram />
        </ContactLink>
        <div className={styles.verticalLine}></div>
      </ul>
    </div>
  )
}

export default Contact
