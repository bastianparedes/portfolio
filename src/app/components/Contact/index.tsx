import React from 'react';

import { AiOutlineLinkedin } from 'react-icons/ai';
import { FiInstagram } from 'react-icons/fi';
import { VscGithubAlt, VscMail } from 'react-icons/vsc';

import ContactLink from './ContactLink';
import styles from './styles.module.scss';
import constants from '../../../../config/constants';

const Contact = () => {
  return (
    <div className={styles.linkContainer}>
      <ul className={styles.linksList}>
        <ContactLink link={constants.CONTACT.LINKS.GITHUB}>
          <VscGithubAlt />
        </ContactLink>
        <ContactLink link={constants.CONTACT.LINKS.EMAIL}>
          <VscMail />
        </ContactLink>
        <ContactLink link={constants.CONTACT.LINKS.LINKEDIN}>
          <AiOutlineLinkedin />
        </ContactLink>
        <ContactLink link={constants.CONTACT.LINKS.INSTAGRAM}>
          <FiInstagram />
        </ContactLink>
        <div className={styles.verticalLine}></div>
      </ul>
    </div>
  );
};

export default Contact;
