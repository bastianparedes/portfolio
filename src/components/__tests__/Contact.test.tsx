import React from 'react'
import Contact from '../Contact'
import { render } from '@testing-library/react'

describe('<Contact />', () => {
  it('should render', () => {
    const { container } = render(<Contact />)
    expect(container).toMatchSnapshot()
  })
})
