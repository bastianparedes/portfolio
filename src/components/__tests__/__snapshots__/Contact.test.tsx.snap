// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<Contact /> should render 1'] = `
<div>
  <div
    class="linkContainer"
  >
    <ul
      class="linksList"
    >
      <li
        class="link"
      >
        <a
          class="link__a"
          href="https://github.com/BastianParedes"
          rel="noreferrer noopener"
          target="_blank"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21.035 5.257c.91 1.092 1.364 2.366 1.364 3.822 0 5.277-3.002 6.824-5.823 7.279.364.637.455 1.365.455 2.093v3.73c0 .455-.273.728-.637.728a.718.718 0 0 1-.728-.728v-3.73a2.497 2.497 0 0 0-.728-2.093l.455-1.183c2.821-.364 5.733-1.274 5.733-6.187 0-1.183-.455-2.275-1.274-3.185l-.182-.727a4.04 4.04 0 0 0 .09-2.73c-.454.09-1.364.273-2.91 1.365l-.547.09a13.307 13.307 0 0 0-6.55 0l-.547-.09C7.57 2.71 6.66 2.437 6.204 2.437c-.273.91-.273 1.91.09 2.73l-.181.727c-.91.91-1.365 2.093-1.365 3.185 0 4.822 2.73 5.823 5.732 6.187l.364 1.183c-.546.546-.819 1.274-.728 2.002v3.821a.718.718 0 0 1-.728.728.718.718 0 0 1-.728-.728V20.18c-3.002.637-4.185-.91-5.095-2.092-.455-.546-.819-1.001-1.274-1.092-.09-.091-.364-.455-.273-.819.091-.364.455-.637.82-.455.91.182 1.455.91 2 1.547.82 1.092 1.639 2.092 4.095 1.547v-.364c-.09-.728.091-1.456.455-2.093-2.73-.546-5.914-2.093-5.914-7.279 0-1.456.455-2.73 1.365-3.822-.273-1.273-.182-2.638.273-3.73l.455-.364C5.749 1.073 7.023.8 9.66 2.437a13.673 13.673 0 0 1 6.642 0C18.851.708 20.216.98 20.398 1.072l.455.364c.455 1.274.546 2.548.182 3.821z"
            />
          </svg>
        </a>
      </li>
      <li
        class="link"
      >
        <a
          class="link__a"
          href="mailto:bastian.p.trabajo@outlook.com"
          rel="noreferrer noopener"
          target="_blank"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M1 3.5l.5-.5h13l.5.5v9l-.5.5h-13l-.5-.5v-9zm1 1.035V12h12V4.536L8.31 8.9H7.7L2 4.535zM13.03 4H2.97L8 7.869 13.03 4z"
              fill-rule="evenodd"
            />
          </svg>
        </a>
      </li>
      <li
        class="link"
      >
        <a
          class="link__a"
          href="https://www.linkedin.com/in/bastianparedes/"
          rel="noreferrer noopener"
          target="_blank"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M847.7 112H176.3c-35.5 0-64.3 28.8-64.3 64.3v671.4c0 35.5 28.8 64.3 64.3 64.3h671.4c35.5 0 64.3-28.8 64.3-64.3V176.3c0-35.5-28.8-64.3-64.3-64.3zm0 736c-447.8-.1-671.7-.2-671.7-.3.1-447.8.2-671.7.3-671.7 447.8.1 671.7.2 671.7.3-.1 447.8-.2 671.7-.3 671.7zM230.6 411.9h118.7v381.8H230.6zm59.4-52.2c37.9 0 68.8-30.8 68.8-68.8a68.8 68.8 0 1 0-137.6 0c-.1 38 30.7 68.8 68.8 68.8zm252.3 245.1c0-49.8 9.5-98 71.2-98 60.8 0 61.7 56.9 61.7 101.2v185.7h118.6V584.3c0-102.8-22.2-181.9-142.3-181.9-57.7 0-96.4 31.7-112.3 61.7h-1.6v-52.2H423.7v381.8h118.6V604.8z"
            />
          </svg>
        </a>
      </li>
      <li
        class="link"
      >
        <a
          class="link__a"
          href="https://www.instagram.com/bastian.p.95/"
          rel="noreferrer noopener"
          target="_blank"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="20"
              rx="5"
              ry="5"
              width="20"
              x="2"
              y="2"
            />
            <path
              d="M16 11.37A4 4 0 1 1 12.63 8 4 4 0 0 1 16 11.37z"
            />
            <line
              x1="17.5"
              x2="17.51"
              y1="6.5"
              y2="6.5"
            />
          </svg>
        </a>
      </li>
      <div
        class="verticalLine"
      />
    </ul>
  </div>
</div>
`;
